store/tikv/2pc.go

```diff
@@ -135,6 +147,26 @@ func newTwoPhaseCommitter(txn *tikvTxn, connID uint64) (*twoPhaseCommitter, erro
 	if len(keys) == 0 {
 		return nil, nil
 	}
+
+	for _, pair := range txn.contracts {
+		mutation, ok := mutations[string(pair.key)]
```

Comment:
What if we first set `k` with constraint, then later set the same `k` without constraint, then we ended up with the constraint we don't want.

Reply:
We'll take the first constraint in this case, and the first constraint is what we want.